pkgs/tools/misc/youtube-dl/default.nix

```diff
@@ -20,8 +20,8 @@ buildPythonPackage rec {
   # to the latest stable release.
   version = "2021.06.06";
 
-  src = fetchurl {
-    url = "https://yt-dl.org/downloads/${version}/${pname}-${version}.tar.gz";
+  src = fetchPypi {
+    inherit pname version;
```

Comment:
`pname`: pypi package name is [`youtube_dl`](https://pypi.org/project/youtube_dl/) but it seems `youtube-dl` redirects there
`version`: pypi package version doesn't have leading zero but it will be normalized

Thus the two attributes are directly inherited.